tests/std/tests/P0798R8_monadic_operations_for_std_optional/test.cpp

```diff
@@ -109,7 +110,23 @@ constexpr bool test() {
     return true;
 }
 
+void test_gh3667() {
+    class unique_exception : public exception {};
+
+    try {
+        optional<int> opt(1);
+        opt.transform([](int) -> int { throw unique_exception{}; });

```

Comment:
There are two codepaths being altered in the product (non-trivial destructor and trivial destructor), so we should really test both codepaths.